packages/react-components/src/IdentityIcon/index.tsx

```diff
@@ -44,18 +44,17 @@ function IdentityIcon ({ className = '', prefix, size = 24, theme, value }: Prop
     }),
     [queueAction, t]
   );
-
-  return (
-    <BaseIdentityIcon
-      Custom={Custom}
-      className={className}
-      onCopy={_onCopy}
-      prefix={prefix}
-      size={size}
-      theme={thisTheme as 'substrate'}
-      value={value}
-    />
-  );
+      return (
+        <BaseIdentityIcon
+          Custom={Custom}
+          className={className}
+          onCopy={_onCopy}
+          prefix={prefix}
+          size={size}
+          theme={isEthereum? 'ethereum' : thisTheme as 'substrate'}
+          value={value.length==42? value+"000000":value}
```

Comment:
Does it still break here? Should just "work"...

Reply:
It does work  without this logic. Good catch!

Reply:
... if not, there would be a bug in the IdentityIcon piece. (It was tested, however only in displaying normal longer addresses... so worried there may be an issue which I didn't catch)

Reply:
The ethereum addresses still show as blank  identicon, but the sr25519 addresses do display

Reply:
Grrrr.... Ok, need to track that one, it should "just work". Ok, not sure how I managed to break it. https://github.com/polkadot-js/ui/blob/master/packages/react-identicon/src/Identicon.tsx#L72-L78

Reply:
I'm getting a lot of 'DRR: Error: Unable to retrieve accountId' error; not sure if it's related.
Also I don't know if this helps but this is the code at /apps/packages/react-components/src/util/toAddress.ts

`export default function toAddress (value?: string | Uint8Array | null, allowIndices = false): string | undefined {
  if (value) {
    try {
      const u8a = isHex(value)
        ? hexToU8a(value)
        : keyring.decodeAddress(value);

      assert(allowIndices || u8a.length === 32 || u8a.length === 20, 'AccountIndex values not allowed');

      if (u8a.length === 20) {
        return ethereumEncode(u8a);
      } else {
        return keyring.encodeAddress(u8a);
      }
    } catch (error) {
      // noop, undefined return indicates invalid/transient
    }
  }

  return undefined;
}`

Reply:
Ahhh... ok... that is a completely separate issue, which I did see initially when I played, but slipped my mind. It comes from here - https://github.com/polkadot-js/apps/blob/master/packages/react-hooks/src/useAccountInfo.ts

Basically on the input it tries to retrieve any relevant info for that address. So on the `AddressMini` component it uses the address returned from it to actually display the icon. so if the above fails (which is non-Eth aware), the actual address comes back as empty.

Reply:
So not sure which one, but some component is playing havoc :)

Reply:
I think the problem is coming from this line (https://github.com/polkadot-js/apps/blob/master/packages/react-components/src/AccountName.tsx#L134) indeed. The info is returned only for non-ethereum addresses

Isn't this where the accountId is derived:
https://github.com/polkadot-js/api/blob/master/packages/api-derive/src/accounts/accountId.ts
?
Because it looks like this could be the cause of the problem

Reply:
Yes, that is def. an issue. The check should be swapped around in that case to rather check for index (<= 8), AccountId can have an override.

Reply:
So how do you want to proceed to solve this issue? This is in the @polkadot/api repo right?

Reply:
Will update the apps deps as soon as api CI is done.

Reply:
thank you

Reply:
@jacogr Updating the dependencies did solve the identicon problem.
However, now that I added an ethereum address to my address book I have a problem when loading the app.
At `/apps/packages/react-api/src/Api.tsx` at the  `// finally load the keyring` step, I have `Error: Expected a valid key to convert, with length 1, 2, 4, 8, 32, 33` even though I modified the code this way: ` isKeyringLoaded() || keyring.loadAll({
    genesisHash: api.genesisHash,
    isDevelopment,
    ss58Format,
    store,
    type: isEthereum? 'ethereum': 'ed25519'
  }, injectedAccounts);`

Also, I noticed that this file is using '@polkadot/ui-keyring' but this package is not in the dependancies of react-api

Reply:
@jacogr reopened

Reply:
created an issue https://github.com/polkadot-js/apps/issues/3983